src/qililab/result/stream_results.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any
+
+import h5py
+import numpy as np
+
+
+def stream_results(shape: tuple, path: str, loops: dict[str, np.ndarray]):
+    """Constructs a StreamArray instance.
+
+    This methods serves as a constructor for user interface of the StreamArray class.
+
+    Args:
+        shape (tuple): results array shape.
+        path (str): path to save results.
+        loops (dict[str, np.ndarray]): dictionary with each loop name in the experiment as key and numpy array as values.
+
+    Examples:
+
+        **1. Executing an experiment and saving the results live time:**
+
+
+        To execute an experiment you first need to define the loops and the values you want to loop for, for example, a Rabi experiment
+        where you will loop over amplitude values, measuring the results out of the amplitude in each of the steps.
+        Then you also need to build, connect, set up, and execute the platform, which together look like:
+
+        .. code-block:: python
+
+            from qibo.models import Circuit
+            import numpy as np
+            import qililab as ql
+
+            # Constants
+            QUBIT = 0
+            M_BUFFER_TIME = 0
+            AMP_VALUES = np.linspace(0, 1, 1000)
+            PLATFORM_PATH = "runcards/galadriel.yml"
+
+            # Building the platform:
+            platform = ql.build_platform(runcard=PLATFORM_PATH)
+
+            # Connecting and setting up the platform:
+            platform.connect()
+            platform.initial_setup()
+            platform.turn_on_instruments()
+
+            # Define circuit
+            circuit = Circuit(QUBIT + 1)
+            circuit.add(ql.Drag(QUBIT, theta=np.pi, phase=0))
+            circuit.add(ql.Wait(QUBIT, M_BUFFER_TIME))
+            circuit.add(gates.M(QUBIT))
```

Comment:
Again, this example is way too long. It contains a lot of information that is not related to this method's functionality.

All the lines I marked above can be deleted. A good example could be:

```python
import qililab as ql

stream_array = ql.stream_results(shape=(5, 2), loops={"amp": AMP_VALUES}, path="results.h5")

with stream_array:
    for i in range(5):
        stream_array[i, 0] = i
```
```pycon
>>> stream_array
# Show output
>>> ql.load_results("results.h5")
# Show output
```

Reply:
The example above is super short, copy-pasteable, and concise (it shows the functionality of the feature).

Reply:
Then, if we want to add more information (for example, related to the slurm magic method) we can add an extra example below:

Reply:
For example:

> .. note::
> 
>     The output of `stream_results` is not picklable, thus it cannot be returned in a SLURM job. In this scenario we suggest copying the array with the data into a different variable and returning that variable instead:
> 
>     .. code-block:: python
> 
>         %%submit_job -o results -d galadriel
>         stream = ql.stream_results(shape=(5, 2), loops={"amp": AMP_VALUES}, path="results.h5")
>         with stream:
>             for i in range(5):
>                 # Here you can execute any algorithm you want
>                 stream[i] = i
>         results = stream.results